test/unit/meta/src/IsTriviallyCopyableTest.cpp

```diff
@@ -0,0 +1,95 @@
+/* Copyright 2019 Benjamin Worpitz
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <alpaka/alpaka.hpp>
+
+#include <alpaka/core/BoostPredef.hpp>
+#if BOOST_COMP_CLANG
+    #pragma clang diagnostic push
+    #pragma clang diagnostic ignored "-Wunused-parameter"
+#endif
+#include <boost/test/unit_test.hpp>
+#if BOOST_COMP_CLANG
+    #pragma clang diagnostic pop
+#endif
+
+#include <functional>
+#if BOOST_LANG_CUDA
+#include <nvfunctional>
+#endif
+
+BOOST_AUTO_TEST_SUITE(meta)
+
+struct A {
+    long long a;
+};
+
+struct B {
+    B(int x): b(x+2) {}
+    B(B const&) = default;
+    int b;
+};
+
+using TriviallyCopyableTypes = std::tuple<char, short, int, long, long long, float, double, A, B, int[], double[], B[], int const, A const, B const []
+#if BOOST_LANG_CUDA
+        ,nvstd::function<void(bool)>
+#endif
+    >;
+
+//-----------------------------------------------------------------------------
+BOOST_AUTO_TEST_CASE_TEMPLATE(
+    isTriviallyCopyable,
+    T,
+    TriviallyCopyableTypes)
+{
+    constexpr bool IsTriviallyCopyableResult =
+        alpaka::meta::IsTriviallyCopyable<
+            T
+        >::value;
+
+    constexpr bool IsTriviallyCopyableReference =
+        true;
+
+    static_assert(
+        IsTriviallyCopyableReference == IsTriviallyCopyableResult,
+        "alpaka::meta::IsTriviallyCopyable failed!");
+}
+
+struct C {
+    C(){}
+    C(C const&) {}
+};
+
+/*struct D {
```

Comment:
Why is this commented out? Could you add a docstring with the reason, please.

Reply:
Not commented out anymore